src/chunk.c

```diff
@@ -2728,10 +2729,15 @@ ts_chunk_exists_with_compression(int32 hypertable_id)
 	init_scan_by_hypertable_id(&iterator, hypertable_id);
 	ts_scanner_foreach(&iterator)
 	{
-		bool isnull =
+		bool isnull_dropped;
+		bool isnull_chunk_id =
 			slot_attisnull(ts_scan_iterator_slot(&iterator), Anum_chunk_compressed_chunk_id);
+		bool dropped = DatumGetBool(
```

Comment:
Also probably need a script to fix this for chunks that are marked as dropped but have compressed_chunk_id still set incorrectly.